src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,747 @@
+
+            # Create a new event loop
+            self._loop = asyncio.new_event_loop()
+            self._loop_thread = threading.Thread(target=self.start_event_loop, args=(self._loop,), daemon=True)
+            self._loop_thread.start()
+
+    def start_event_loop(self, loop):
+        asyncio.set_event_loop(loop)
+        loop.run_forever()
+
+    async def guess_samples_per_pixel(self, req_dt: float) -> float:
+        """
+        Samples per pixel (SPP) is the number of samples that are measured at each pixel and averaged.
+        Calculate the best guess for the requested dwell time using a range of SPP apply the 50% guess method.
+        :param req_dt: The requested dwell time in microseconds.
+        :return: Closest calculated samples per pixel value.
+        """
+        best_spp = 1.0
+        req_dt *= 1e6
+
+        for i in range(10):
+            # OS, channels, samples, delay
+            async with self.client:
+                dt = await self.ebic_info_node.call_method(f"{NAMESPACE_INDEX}:calculate_dwell_time", 0, 1, best_spp, 0)
+            if math.isclose(dt, req_dt, abs_tol=1e-3):  # abs tolerance of 1 ns
+                logging.debug("Samples per pixels match found for requested dwell time")
+                break
+            elif dt > req_dt:
+                best_spp /= dt / req_dt
+            elif dt < req_dt:
+                best_spp *= req_dt / dt
+
+        return best_spp
+
+    async def _read_controller_state(self):
+        async with self.client:
+            ret_val = await self._ebic_state_node.read_value()
+        return ret_val
+
+    def _get_controller_state(self):
+        f = asyncio.run_coroutine_threadsafe(self._read_controller_state(), self._loop)
+        return f.result()
+
+    def set_controller_state(self, new_state):
+        f = asyncio.run_coroutine_threadsafe(self._set_controller_state(new_state), self._loop)
+        f.result()
+
+    async def _set_controller_state(self, new_state):
+        async with self.client:
+            await self._ebic_controller_node.call_method(f"{NAMESPACE_INDEX}:set_controller_state", new_state)
+
+    def get_scan_time(self, dt, res_fast, res_slow) -> float:
+        f = asyncio.run_coroutine_threadsafe(self._get_scan_time(dt, res_fast, res_slow) , self._loop)
+        return f.result()
+
+    async def _get_scan_time(self, dt, res_fast, res_slow) -> float:
+        async with self.client:
+            # dt pf, ps
+            st = await self.ebic_info_node.call_method(f"{NAMESPACE_INDEX}:calculate_scan_time", dt, res_fast, res_slow)
+            return st
+
+    def start_scan(self, req_dt, oversampling, channels, delay, p_fast, p_slow, sim=False):
+        f = asyncio.run_coroutine_threadsafe(self._start_scan(req_dt, oversampling, channels, delay, p_fast, p_slow, sim), self._loop)
+        return f
+
+    async def _start_scan(self, req_dt, oversampling, channels, delay, p_fast, p_slow, sim=False):
```

Comment:
Please call this function `_start_trigger_scan`, as it's the name of the function it wraps.